On The Map/Controller/MapKitViewController.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  MapKitViewController.swift
+//  On The Map
+//
+//  Created by Michal Hus on 2/25/20.
+//  Copyright © 2020 Michal Hus. All rights reserved.
+//
+
+import UIKit
+import MapKit
+
+class MapKitViewController: UIViewController, MKMapViewDelegate {
+    
+    var students:[Student] = []
+
+    @IBOutlet weak var mapView: MKMapView!
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        mapView.delegate = self
+        self.pinLocatinDisplay()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(true)
+        Client.getStudentsLocation { (response, error) in
+            guard let response = response, error == nil else {
+                return
+            }
+            self.refetchLocations(response: response)
+        }
+    }
+    
+    @IBAction func refreshStudentLocation(_ sender: Any) {
+        Client.getStudentsLocation { (response, error) in
+            guard let response = response, error == nil else {
+                return
+            }
+            self.refetchLocations(response: response)
+        }
+    }
+    
+    // MARK: - MKMapViewDelegate
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        
+        guard annotation is MKPointAnnotation else { return nil }
+
+        let reuseId = "pin"
+        
+        var pinView = mapView.dequeueReusableAnnotationView(withIdentifier: reuseId) as? MKPinAnnotationView
+        
+        if pinView == nil {
+            pinView = MKPinAnnotationView(annotation: annotation, reuseIdentifier: reuseId)
+            pinView!.canShowCallout = true
+            pinView!.pinTintColor = .red
+            pinView!.rightCalloutAccessoryView = UIButton(type: .detailDisclosure)
+        }
+        else {
+            pinView!.annotation = annotation
+        }
+        return pinView
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        if control == view.rightCalloutAccessoryView {
+            guard let toOpen = view.annotation?.subtitle else {
+                errorAlertMessage(title: "Error", message: "There is something wrong with the link.")
+                return
+            }
+            UIApplication.shared.open(URL(string: toOpen!)!, options: [:], completionHandler: nil)
```

Comment:
probably wanna unwrap `toOpen` and the URL you make from it in a guard rather than force unwrapping here

Reply:
Seems like I should minimize force unwrapping to a minimum which is a valid concern.